src/App.tsx

```diff
@@ -1,26 +1,105 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {useState} from 'react';
+
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+interface Product {
+    category: string;
+    price: string;
+    stocked: boolean;
+    name: string;
+}
+
+interface AllProducts {
+    products: Array<Product>;
+};
+
+export const App: React.FC<AllProducts> = ({products}) => {
+    const [searchText, setText] = useState("")
```

Comment:
setState values should have names like [loading, setLoading], [isAuth, setIsAuth], [myCoolName, setMyCoolName], etc
So rename to 
```jsx
const [searchText, setSearchText] = useState('');
```
or to:
```jsx
const [text, setText] = useState('');
```